rules/aws_s3_bucket_name.go

```diff
@@ -96,12 +96,8 @@ func (r *AwsS3BucketNameRule) Check(runner tflint.Runner) error {
 			Description: "Bucket names can consist only of lowercase letters, numbers, dots (.), and hyphens (-).",
 		},
 		{
-			Regexp:      *regexp.MustCompile("^[^a-z0-9]"),
-			Description: "Bucket names must begin with a lowercase letter or number.",
-		},
-		{
-			Regexp:      *regexp.MustCompile("[^a-z0-9]$"),
-			Description: "Bucket names must end with a lowercase letter or number.",
+			Regexp:      *regexp.MustCompile("^[^a-z0-9]|[^a-z0-9]$"),
+			Description: "Bucket names must begin and end with a lowercase letter or number.",
```

Comment:
The alternation operator could also be used here to merge these two rules in one, just as the docs.